src/Storages/System/StorageSystemTables.cpp

```diff
@@ -239,6 +245,38 @@ class TablesBlockSource : public ISource
     String getName() const override { return "Tables"; }
 
 protected:
+    NameToNameMap getSelectParamters(const StorageMetadataPtr & metadata_snapshot)
+    {
+        const SelectQueryDescription & query_description = metadata_snapshot->getSelectQuery();
+        ASTPtr inner_query = query_description.inner_query;
+        if (!inner_query || !inner_query->as<ASTSelectWithUnionQuery>())
+            return {};
+
+        return inner_query->as<ASTSelectWithUnionQuery>()->getQueryParameters();
+    }
+
+    void FillParametralizedViewData(MutableColumns & columns, const StoragePtr & table, size_t & res_index, size_t & src_index)
+    {
+        if (src_index == 14 && columns_mask[src_index++])
+        {
+            if (table)
+            {
+                StorageMetadataPtr metadata_snapshot = table->getInMemoryMetadataPtr();
```

Comment:
As this is only for parameterized view, we can check if StoragePtr is parameterized view & only get query parameters for it. 

Something like 
`if (table.get()->isView() && table->as<StorageView>() && table->as<StorageView>()->isParameterizedView())`

and assert that they do have query parameters

Reply:
The reason for this condition is that system.tables requires each column to have the same length.
https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/System/StorageSystemTables.cpp#L242

Your conditon cause a crash because other columns in the table, if they are not part of a parameterized view, won’t insert default values.

For example, this query will make server crash.
SELECT name, engine, parametrized_view_paramters FROM system.tables;

I updated code for this case. Please help to recheck.